sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/SessionService.java

```diff
@@ -195,42 +221,49 @@ private static JCSMPProperties overrideConnectorProperties(
         LOG.warn(
             "SolaceIO.Write: Overriding MESSAGE_CALLBACK_ON_REACTOR to false since"
                 + " HIGHER_THROUGHPUT mode was selected");
+        props.setProperty(JCSMPProperties.MESSAGE_CALLBACK_ON_REACTOR, false);
       }
 
-      props.setProperty(JCSMPProperties.MESSAGE_CALLBACK_ON_REACTOR, false);
-
       Integer ackWindowSize = props.getIntegerProperty(JCSMPProperties.PUB_ACK_WINDOW_SIZE);
       if ((ackWindowSize != null && ackWindowSize != BATCHED_PUB_ACK_WINDOW)) {
         LOG.warn(
             String.format(
                 "SolaceIO.Write: Overriding PUB_ACK_WINDOW_SIZE to %d since"
                     + " HIGHER_THROUGHPUT mode was selected",
                 BATCHED_PUB_ACK_WINDOW));
+        props.setProperty(JCSMPProperties.PUB_ACK_WINDOW_SIZE, BATCHED_PUB_ACK_WINDOW);
       }
-      props.setProperty(JCSMPProperties.PUB_ACK_WINDOW_SIZE, BATCHED_PUB_ACK_WINDOW);
-    } else {
+
+    } else if (mode == SolaceIO.SubmissionMode.LOWER_LATENCY) {
       // Send from the same thread where the produced is being called. This offers the lowest
```

Comment:
Do you really want to remodel the properties of Solace's API in this connector using different names? Can't this connector expose a `.withProperties(java.util.Properties)` and convert that using `JCSMPProperties.fromProperties(java.util.Properties)`?

Reply:
It is not a single property, it is a combination of properties that work together. In this case, it is actually two properties. I guess I could add another mode that "go fully custom", where you control all the properties. The connector actually exposes all the properties, and you can set them to whatever value, but for two specific properties (publication ack window and message callback on reactor) it makes a "sensible" choice for you. The performance of the writer is heavily impacted by those values.

Reply:
I have added a new method that lets the user to "go fully custom". Please notice that when you create a custom `SessionService` you have full control over all the properties. With this new submission method, the connector will not override any property. So now there are three methods: high throughput (overrides 2 properties), low latency (overrides same 2 properties with different values), custom (user properties are left intact). I will resolve now this comment, as I will be submitting a new commit with those changes soon.